### DISCUSSION TITLE
[Bug][Master] there exists some problems in checking task dependency

### DISCUSSION THREAD

**Discussion by legen618** [apache/dolphinscheduler#6776] [Ideas]
**Describe the bug**
There exists some problems in Master-Server executing logic. Some nodes could execute normally though their depenpency is not satisfied.  Just like the screenshots below, if you get the same work flow, and you will find the result is wrong.

**Screenshots**
As you can see, task B shouldn't be executed which is the opposite of the result. When task X has condition-type node after, the tasks that depend on X will be probably executed.
![微信图片_20200727231746](https://user-images.githubusercontent.com/37466734/88561201-ac56db80-d061-11ea-9950-69cbeb4d5958.png)

**Which version of Dolphin Scheduler:**
 -[1.3.2-snapshot]

**Probable reason**
- it's wrong because of the code below in the red rectangle.
![微信图片_20200727231733](https://user-images.githubusercontent.com/37466734/88561285-c8f31380-d061-11ea-966f-1b6a246021fc.png)

**Comment by xingchun-chen** [comment]
Whether the X task status of the custom parameter configuration in the condition node is successful or failed, can you take a screenshot of the configuration of the condition node

**Comment by legen618** [comment]
> Whether the X task status of the custom parameter configuration in the condition node is successful or failed, can you take a screenshot of the configuration of the condition node

**Whether the X task status of the custom parameter configuration in the condition node is successful or failed, the result is the same.**

**Comment by legen618** [comment]
**some similar problems are as follows:**
![bug](https://user-images.githubusercontent.com/37466734/88620858-97b22c00-d0d1-11ea-846d-07da3091d076.PNG)
![bug2](https://user-images.githubusercontent.com/37466734/88620868-a0a2fd80-d0d1-11ea-985d-3b820ab35a2e.PNG)

**I'll fix it later this afternoon.**

**Comment by petermenglei** [comment]
I have the same issue here, after upgrade to version 1.3.0, **all new created process** have dependency issue (checked the configuration in table, No pretasks configured). anyone know to get around of this issue, even temporily?

**Comment by lenboo** [comment]
> I have the same issue here, after upgrade to version 1.3.0, **all new created process** have dependency issue (checked the configuration in table, No pretasks configured). anyone know to get around of this issue, even temporily?

you can submit another issue about 'no pretasks configured'.

**Comment by xingchun-chen** [comment]
1. As shown in the figure below, node 5 should not be executed
![image](https://user-images.githubusercontent.com/55787491/90210443-8df22d80-de20-11ea-9bb4-18ae05f979af.png)
2. As shown in the figure below, node 9 and node10 should not be executed
![image](https://user-images.githubusercontent.com/55787491/90210523-be39cc00-de20-11ea-8af2-dd5a19a31b81.png)
3.As shown in the figure below, "嵌套条件"node is not set to fail
![image](https://user-images.githubusercontent.com/55787491/90210996-1a512000-de22-11ea-9041-b206e839e73d.png)

**Comment by lenboo** [comment]
i think we have some points about issue need to be discussed, so i will start a new discussion in mail list.
